Docs for all Rust services (#116)
MiSArch/docs: docs/docs/dev-manuals/services/wishlist.mdx

import CustomImage from "@site/src/components/CustomImage";

# Wishlist Service

The wishlist service provides the bounded context `Wishlist`.
It is responsible for managing wishlists of users.

## Domain Model

> // TODO: Niklas

Reviewer: What exactly is @nk-coding supposed to do?

Reviewer 2: it means that I agreed a while ago on drawing the hylimo diagrams for the other services if necessary and I did so with a7cfffc929ee8986ec156d9d3ce7e002ec132e21

- **Language**: Rust
- **Framework**: [axum](https://docs.rs/axum/latest/axum/)
- **GraphQL Library**: [async_graphql](https://docs.rs/async-graphql/latest/async_graphql/)
- **Database**: MongoDB

## Repository Structure

The repository is structured as follows:

<div className="repository-structure">
- `/src`: Source code of the service
  - `authentication.rs` Header authentication logic of service
  - `base_connection.rs` Generic GraphQL connection
  - `foreign_types.rs` Datatypes which are not part of the services bounded context
  - `http_event_service.rs` Handling of incoming events
  - `main.rs` Service execution and schema generation
  - `mutation_input_structs.rs` GraphQL input datatypes for mutations
  - `mutation.rs` GraphQL mutations and associated functions
  - `order_datatypes.rs` GraphQL datatypes for connection element ordering
  - `product_variant_connection.rs` GraphQL connection of product variants
  - `query.rs` GraphQL queries and associated functions
  - `user.rs` User with additional GraphQL field(s) provided by this service
  - `wishlist_connection.rs` GraphQL connection of wishlist
  - `wishlist.rs` Wishlist datatype
</div>

## Defined Events

> This section lists events that are defined by the address service and can be used by other services.

Reviewer: ```suggestion
This section lists events that are defined by the wishlist service and can be used by other services.
```

### Published Defined Events

n/a

### Subscribed Defined Events

n/a

## Imported Events

> This section lists events defined by other services that are used by the address service.

Reviewer: ```suggestion
This section lists events defined by other services that are used by the wishlist service.
```

### Published Imported Events

n/a

### Subscribed Imported Events

> - [user/user/created](/docs/docs/dev-manuals/services/user#userusercreated): Used to validate that wishlists belong to an existing users

Reviewer: ```suggestion
- [user/user/created](/docs/docs/dev-manuals/services/user#userusercreated): Used to validate that wishlists belong to an existing user
```

> - [catalog/product-variant/created](/docs/docs/dev-manuals/services/catalog#catalogproduct-variantcreated): Used to validate product variants of in wishlists

Reviewer: ```suggestion
- [catalog/product-variant/created](/docs/docs/dev-manuals/services/catalog#catalogproduct-variantcreated): Used to validate that product variants contained in wishlists exist
```

The repository is structured as follows:

<div className="repository-structure">
 - `/src`: Source code of the service - `authentication.rs` Header authentication logic of service -
 `base_connection.rs` Generic GraphQL connection - `foreign_types.rs` Datatypes which are not part of the services
 bounded context - `http_event_service.rs` Handling of incoming events - `main.rs` Service execution and schema
 generation - `mutation_input_structs.rs` GraphQL input datatypes for mutations - `mutation.rs` GraphQL mutations and
 associated functions - `order_datatypes.rs` GraphQL datatypes for connection element ordering -
 `product_variant_connection.rs` GraphQL connection of product variants - `query.rs` GraphQL queries and associated
 functions - `user.rs` User with additional GraphQL field(s) provided by this service - `wishlist_connection.rs` GraphQL

>  connection of wishlist - `wishlist.rs` Wishlist datatype

Reviewer: The formatting seems to have gone haywire, please restore the old value
